compiler warning with Visual Studio 2019 [5111]

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.20.27508\include\xstring(2268): warning C4244: 'argument': conversion from 'wchar_t' to 'const _Elem', possible loss of data [C:\J\workspace\ci_windows_vs2019\ws\build\fastrtps\src\cpp\fastrtps.vcxproj]
          with
          [
              _Elem=char
          ] (compiling source file C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp)
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.20.27508\include\xstring(2279): note: see reference to function template instantiation 'void std::basic_string<char,std::char_traits<char>,std::allocator<char>>::_Construct<wchar_t*>(_Iter,const _Iter,std::input_iterator_tag)' being compiled
          with
          [
              _Iter=wchar_t *
          ] (compiling source file C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp)
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.20.27508\include\xstring(2279): note: see reference to function template instantiation 'void std::basic_string<char,std::char_traits<char>,std::allocator<char>>::_Construct<wchar_t*>(_Iter,const _Iter,std::input_iterator_tag)' being compiled
          with
          [
              _Iter=wchar_t *
          ] (compiling source file C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp)
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.20.27508\include\xstring(2259): note: see reference to function template instantiation 'void std::basic_string<char,std::char_traits<char>,std::allocator<char>>::_Construct<wchar_t*>(const _Iter,const _Iter,std::forward_iterator_tag)' being compiled
          with
          [
              _Iter=wchar_t *
          ] (compiling source file C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp)
  DynamicTypePtr.cpp
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.20.27508\include\xstring(2259): note: see reference to function template instantiation 'void std::basic_string<char,std::char_traits<char>,std::allocator<char>>::_Construct<wchar_t*>(const _Iter,const _Iter,std::forward_iterator_tag)' being compiled
          with
          [
              _Iter=wchar_t *
          ] (compiling source file C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp)
  C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp(1117): note: see reference to function template instantiation 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::_String_iterator<std::_String_val<std::_Simple_types<_Elem>>>,void>(_Iter,_Iter,const _Alloc &)' being compiled
          with
          [
              _Elem=wchar_t,
              _Iter=std::_String_iterator<std::_String_val<std::_Simple_types<wchar_t>>>,
              _Alloc=std::allocator<char>
          ]
  C:\J\workspace\ci_windows_vs2019\ws\src\eProsima\Fast-RTPS\src\cpp\types\DynamicData.cpp(1117): note: see reference to function template instantiation 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::_String_iterator<std::_String_val<std::_Simple_types<_Elem>>>,void>(_Iter,_Iter,const _Alloc &)' being compiled
          with
          [
              _Elem=wchar_t,
              _Iter=std::_String_iterator<std::_String_val<std::_Simple_types<wchar_t>>>,
              _Alloc=std::allocator<char>
          ]
```

--- a/src/cpp/types/DynamicData.cpp
+++ b/src/cpp/types/DynamicData.cpp
@@ -1115,11 +1115,11 @@ void DynamicData::GetValue(std::string& sOutValue, MemberId id /*= MEMBER_ID_INV
     {
         wchar_t value(0);
         GetChar16Value(value, id);
-		using convert_type = std::codecvt_utf8<wchar_t>;
-		std::wstring_convert<convert_type, wchar_t> converter;
+        using convert_type = std::codecvt_utf8<wchar_t>;
+        std::wstring_convert<convert_type, wchar_t> converter;
         std::wstring temp = L"";
         temp += value;
-		sOutValue = converter.to_bytes(temp);
+        sOutValue = converter.to_bytes(temp);
     }
     break;
     case TK_BOOLEAN:
@@ -1143,11 +1143,11 @@ void DynamicData::GetValue(std::string& sOutValue, MemberId id /*= MEMBER_ID_INV
     break;
     case TK_STRING16:
     {
-		using convert_type = std::codecvt_utf8<wchar_t>;
-		std::wstring_convert<convert_type, wchar_t> converter;
+        using convert_type = std::codecvt_utf8<wchar_t>;
+        std::wstring_convert<convert_type, wchar_t> converter;
         std::wstring value;
         GetWstringValue(value, id);
-		sOutValue = converter.to_bytes(value);
+        sOutValue = converter.to_bytes(value);
     }
     break;
     case TK_ENUM:

--- a/src/cpp/types/DynamicData.cpp
+++ b/src/cpp/types/DynamicData.cpp
@@ -23,6 +23,9 @@
 #include <fastrtps/log/Log.h>
 #include <fastcdr/Cdr.h>
 
+#include <locale>
+#include <codecvt>
+
 namespace eprosima {
 namespace fastrtps {
 namespace types {
@@ -1112,9 +1115,11 @@ void DynamicData::GetValue(std::string& sOutValue, MemberId id /*= MEMBER_ID_INV
     {
         wchar_t value(0);
         GetChar16Value(value, id);
+		using convert_type = std::codecvt_utf8<wchar_t>;
+		std::wstring_convert<convert_type, wchar_t> converter;
         std::wstring temp = L"";
         temp += value;
-        sOutValue = std::string(temp.begin(), temp.end());
+		sOutValue = converter.to_bytes(temp);
     }
     break;
     case TK_BOOLEAN:
@@ -1138,9 +1143,11 @@ void DynamicData::GetValue(std::string& sOutValue, MemberId id /*= MEMBER_ID_INV
     break;
     case TK_STRING16:
     {
+		using convert_type = std::codecvt_utf8<wchar_t>;
+		std::wstring_convert<convert_type, wchar_t> converter;
         std::wstring value;
         GetWstringValue(value, id);
-        sOutValue = std::string(value.begin(), value.end());
+		sOutValue = converter.to_bytes(value);
     }
     break;
     case TK_ENUM:

